### PR TITLE
#137156961 Make Api endpoints restful

### PR DESCRIPTION
**Pull request by fuzzybrows**
#### What does this PR do?
Enables Restful API endpoints for Applications

 #### Description of Task to be completed?
Provides the under listed API endpoints
- User
  - Ordered Products ("/api/ordered_products/:userEmail")
    - Product Name
    - Product Price
    - Date of Purchase
    - Order Status
    - Delivery Address
  - Cancelled Orders  ("/api/cancelled_orders/:userEmail")
    - Product Name
    - Product Price
    - Date of Purchase
    - Cancellation Date
    - Delivery Address
  - Processed Orders ("/api/processed_orders/:userEmail")
    - Product Name
    - Product Price
    - Date of Purchase
    - Delivery Address

- Administrator
  - Shops ("/api/shops")
    - Shop Name
    - Shop ID
    - Owner's Email
    - LastUpdated
  - Users ("/api/users")
    - User's Email
    - User's Name
    - Date of Registration
    - Verified

- Public 
  - Products ("/api/products")
    - Product Name
    - Product ID
    - Vendor
    - Price
    - Quantity In Stock
 #### How should this be manually tested?
append the following url to hostname
test email: 'oreoluwa.banwo@andela.com'
http://banasko-rc-pr10.thetechlads.tk:8000/api/products
http://banasko-rc-pr10.thetechlads.tk:8000/api/users
http://banasko-rc-pr10.thetechlads.tk:8000/api/shops
http://banasko-rc-pr10.thetechlads.tk:8000/api/ordered_products/${userEmail}
http://banasko-rc-pr10.thetechlads.tk:8000/api/processed_orders/${userEmail}
http://banasko-rc-pr10.thetechlads.tk:8000/api/cancelled_orders/${userEmail}

 #### Any background context you want to provide?
Api endpoints are served up with an Express Server each wrapping a GraphQL query

 #### What are the relevant pivotal tracker stories?
#137156961 Make the API endpoints RESTful

### CODE REVIEW

File: server/api/custom/graphqlSchema.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    inventoryQuantity: { type: GraphQLInt}
+  })
+});
+
+const UsersType = new GraphQLObjectType({
+  name: "Users",
+  description: "Returns select fields for all Users",
+  fields: () => ({
+    id: { type: GraphQLString },
+    createdAt: {type: GraphQLString},
```
Comment (fuzzybrows) [inline server/api/custom/graphqlSchema.js]: Be consistent with spacing within the Curly braces

File: server/api/custom/graphqlSchema.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    emails: {
+      type: GraphQLString,
+      resolve: (obj) => {
+        if (!obj.emails[0]) {
+          return "No Email Specified";
```
Comment (fuzzybrows) [inline server/api/custom/graphqlSchema.js:41]: Update conditional to '(obj.emails.length)'

File: server/api/custom/graphqlSchema.js [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+      description: "Display Shops",
+      resolve: () => {
+        return Shops.find().fetch();
+      }
+    },
+    orders: {
+      type: new GraphQLList(OrdersType),
+      description: "Display Orders",
+      args: {
+        emailID: {type: GraphQLString},
```
Comment (fuzzybrows) [inline server/api/custom/graphqlSchema.js]: Reduce nested If conditions to a simple use of if, else if and else statements.

File: server/api/custom/graphqlServer.js [hunk 4]
```diff
@@ -0,0 +1,216 @@
+import express from "express";
+import GraphQLHTTP from "express-graphql";
+import schema from "./graphqlSchema";
+import axios from "axios";
+// const schema = require("./graphqlSchema");
```
Comment (fuzzybrows) [inline server/api/custom/graphqlServer.js]: to remove

File: server/api/custom/graphqlServer.js [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+app.get("/api/products", Meteor.bindEnvironment((request, response) => {
+  axios.post(`http://${request.headers.host}/graphql`,
+    {query: "{ products {title _id vendor price inventoryQuantity}}"},
+    {
+      headers: {
+        "Content-Type": "application/json"
+      }
+    })
+    .then(function (res) {
```
Comment (fuzzybrows) [inline server/api/custom/graphqlServer.js]: change to es6 arrow function

File: package.json [hunk 6]
```diff
@@ -82,7 +90,8 @@
     "tether-drop": "^1.4.2",
     "tether-tooltip": "^1.2.0",
     "transliteration": "^1.1.9",
-    "url": "^0.11.0"
+    "url": "^0.11.0",
+    "webpack": "^1.14.0"
```
Comment (adebayo-maborukoje) [inline package.json]: why are we bringing in webpack?

File: server/api/custom/graphqlSchema.js [hunk 7]
```diff
@@ -0,0 +1,216 @@
+import {
+ GraphQLSchema,
```
Comment (adebayo-maborukoje) [inline server/api/custom/graphqlSchema.js]: This should be a 2 space indentation I can only see one here

File: server/api/custom/graphqlSchema.js [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+    },
+    deliveryAddress: {
+      type: ShippingAddress,
+      resolve: (obj) => {
+        return obj.shipping[0].address;
+      }
+    }
+  })
+});
+
```
Comment (adebayo-maborukoje) [inline server/api/custom/graphqlSchema.js:162]: remove one space here. (one should be enough)

File: server/api/custom/graphqlServer.js [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+        }
+      }`
+    },
+    {
+      headers: {
+        "Content-Type": "application/json"
+      }
+    })
+    .then((res) => {
+      response.json(res.data);
```
Comment (adebayo-maborukoje) [inline server/api/custom/graphqlServer.js]: I do not see you return any status code here and in most of the other endpoints.

Kindly rectify that.

File: server/api/custom/graphqlServer.js [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+      }`
+    },
+    {
+      headers: {
+        "Content-Type": "application/json"
+      }
+    })
+    .then((res) => {
+      if (res.data.data.orders.length) {
+        response.json(res.data);
```
Comment (adebayo-maborukoje) [inline server/api/custom/graphqlServer.js]: this should be `response.status().json()`

File: package.json (hunk 6 above)
Comment (fuzzybrows) [reply package.json]: I'm not sure but i think it's a dependency of a module we installed. I've removed it and will test

File: server/api/custom/graphqlServer.js [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+    .then((res) => {
+      response.status(200).json(res.data);
+    })
+    .catch((error) => {
+      response.status(400).send(error);
+    });
+});
+
+app.get("/api/users", (request, response) => {
+  axios.post(`http://${request.headers.host}/graphql`,
```
Comment (andela-oakinseye) [inline server/api/custom/graphqlServer.js:31]: What would happen if the server runs over SSL on line 15?

**Review by adebayo-maborukoje — APPROVED** [review]

**Pull request merged**